tests/sentry/roles/test_manager.py

```diff
@@ -0,0 +1,147 @@
+from unittest import mock
+
+from sentry.roles import RoleManager, default_manager
+from sentry.testutils import TestCase
+
+
+class RoleManagerTest(TestCase):
+    @staticmethod
+    def _assert_entry_is(manager: RoleManager, org_role: str, team_role: str) -> None:
+        assert manager.get_entry_role(org_role).id == team_role
+
+    def test_default_manager(self):
+        assert default_manager.get_all()
+        assert len(default_manager.get_choices()) == len(default_manager.get_all())
+        assert default_manager.get_top_dog().id == "owner"
+
+        assert default_manager.can_manage("owner", "manager")
+        assert default_manager.can_manage("owner", "member")
+        assert default_manager.can_manage("manager", "member")
+
+        self._assert_entry_is(default_manager, "member", "contributor")
+        self._assert_entry_is(default_manager, "admin", "admin")
+        self._assert_entry_is(default_manager, "manager", "admin")
+        self._assert_entry_is(default_manager, "owner", "admin")
+
+    TEST_ORG_ROLES = [
+        {"id": "peasant", "name": "Peasant"},
+        {"id": "baron", "name": "Baron"},
+        {"id": "earl", "name": "Earl"},
+        {"id": "duke", "name": "Duke"},
+        {"id": "monarch", "name": "Monarch"},
+    ]
+
+    TEST_TEAM_ROLES = [
+        {"id": "private", "name": "Private", "is_entry_role_for": "peasant"},
+        {"id": "sergeant", "name": "Sergeant", "is_entry_role_for": "earl"},
+        {"id": "lieutenant", "name": "Lieutenant"},
+        {"id": "captain", "name": "Captain", "is_entry_role_for": "monarch"},
+    ]
```

Comment:
do these roles test behavior that isn't introduced by our config'd roles? if so, what do you think about just using our roles? i am slow to understand the tests with the medieval names :P

Reply:
I do like having a set of tests that exercise `RoleManager` independently of what's in `sentry/conf/server.py`. (It's also important to test the actual config'd values, which I think `test_default_manager` does adequately.)

The other reason I really wanted these fake roles is that I needed at least three team roles to fully cover the code (in case of non-default configs). Having Team Admin be the highest team role left a few paths untested, around things being accessible by default to the org "top dog", and so on.